openstack/identity/identity.go

```diff
@@ -252,6 +252,22 @@ func (h Handler) validateToken(r *http.Request) bool {
 	vars := mux.Vars(r)
 	tenant := vars["tenant"]
 
+	// tenant may not exists on vars due to API endpoints lacking of tenantID
+	// such as the image service, for this case we need to retrieve tenant
+	// from the given X-Auth-Token
+	if tenant == "" {
+		token := r.Header["X-Auth-Token"]
+		if token == nil {
```

Comment:
This test should be

```
if len(token) == 0 {
   return false
}
```
Otherwise token[0] might below might panic for example if token was []string{}.